pkg/registry/apis/provisioning/finalizers.go

```diff
@@ -0,0 +1,142 @@
+package provisioning
+
+import (
+	"sort"
+	"strings"
+
+	"golang.org/x/net/context"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/grafana/grafana-app-sdk/logging"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	dashboards "github.com/grafana/grafana/pkg/apis/dashboard"
+	folders "github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/repository"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
+)
+
+// Remove everything this repo created
+const REMOVE_ORPHAN_RESOURCE = "remove-orphan-resources"
+
+// Remove the metadata for anything this repo created
+const RELEASE_ORPHAN_RESOURCE = "release-orphan-resources"
+
+// Calls the "OnDelete" function for resource
+const CLEANUP_FINALIZER = "cleanup"
+
+type finalizer struct {
+	lister resources.ResourceLister
+	client *resources.ClientFactory
+}
+
+func (f *finalizer) process(ctx context.Context,
+	repo repository.Repository,
+	finalizers []string,
+) error {
+	logger := logging.FromContext(ctx)
+
+	for _, finalizer := range finalizers {
+		switch finalizer {
+		// Unless a finalizer is set, not callback will happen
+		case CLEANUP_FINALIZER:
+			hooks, ok := repo.(repository.RepositoryHooks)
+			if ok {
+				if err := hooks.OnDelete(ctx); err != nil {
+					logger.Warn("Error running deletion hooks", "err", err)
```

Comment:
shouldn't this one return the error?

Reply:
ok, let's leave it as it is. I opened this issue a while ago about the entire behavior of the controller: https://github.com/grafana/git-ui-sync-project/issues/49